numpy/core/src/multiarray/flagsobject.c

```diff
@@ -755,15 +740,9 @@ arrayflags_richcompare(PyObject *self, PyObject *other, int cmp_op)
     }
 
     if (PyObject_TypeCheck(other, &PyArrayFlags_Type)) {
-        cmp = arrayflags_compare((PyArrayFlagsObject *)self,
-                                 (PyArrayFlagsObject *)other);
-
-        if (cmp_op == Py_EQ) {
-            result = (cmp == 0) ? Py_True : Py_False;
-        }
-        else if (cmp_op == Py_NE) {
-            result = (cmp != 0) ? Py_True : Py_False;
-        }
+        cmp = ((PyArrayFlagsObject*) self)->flags ==
+              ((PyArrayFlagsObject*) other)->flags;
+        result = ((cmp == 1) == (cmp_op == Py_EQ)) ? Py_True : Py_False;
```

Comment:
I prefer the long version to this one line statement I think. I guess this might micro-optimize, but I will assume that either the compiler manages, and it does not matter anyway.
Not sure what others think, I would indent the second line of the `==` above a bit more (and maybe add python style brackets around the whole thing).

Reply:
I'm happy to back out this "simplification", It looked like
```
if (a) {
  return 1;
}
return 0;
```

which I just really dislike

Reply:
Perhaps `PyBool_FromLong` instead of the ternary would help here.

Reply:
> Not sure what others think, I would indent the second line of the `==` above a bit more (and maybe add python style brackets around the whole thing).

I'm not sure what you mean by python style brackets, is that pair of parens that wrap the whole expression?

Reply:
I didn't find anything elegant so we're back to something closer to the original.

Reply:
Never mind about it, but to clarify, I meant parentheses around the whole statement since it is multi-line (python style, since it does not use the semicolon).